rflx/generator/generator.py

```diff
@@ -2402,11 +2402,10 @@ def __create_valid_predecessor_function(
             private=[
                 ExpressionFunctionDeclaration(
                     specification,
-                    Case(
-                        Variable("Fld"),
-                        [
-                            (
-                                Variable(f.affixed_name),
+                    Or(
+                        *[
+                            And(
+                                Equal(Variable("Fld"), Variable(f.affixed_name)),
                                 Or(
```

Comment:
We decided to not use the changed `Valid_Predecessor` function, as it increased the binary size (significantly in case of SPDM). Is it strictly required for provability?

Reply:
I don't know, I need to try without.

Reply:
It turns out this change *does* help proof, though it's likely just a coincidence. The new SPARK has trouble with the postcondition of Field_Last, and indirectly this post depends on Valid_Predecessor, and for some reason with this change it goes through. I am trying some other modifications to see if I can make it work without this change.

Reply:
I have reverted this commit. To prove the postcondition of `Field_Last`, I have replaced some `mod` operations by `rem` which yields the same result if both operands are positive (or zero).

Reply:
Is it generally recommendable to use `rem` instead of `mod` if just natural numbers are used?

Reply:
Honestly, I don't know. If only non-negative numbers are involved they both yield the same result, so it doesn't matter. Here this is more an oddity of our proof encoding, which makes `mod` slightly less efficient for proof than `rem`, because `mod` is expressed in terms of `rem`.

Note that last week I played with this a bit and found that if I replaced all/most `mod` by `rem` there were unproved VCs again. So I wouldn't push to replace all.

Reply:
I see, then let's keep it as it is.